src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -242,6 +242,14 @@ typedef struct otSysInfraNetIfAddressCounters
  */
 void otSysCountInfraNetifAddresses(otSysInfraNetIfAddressCounters *aAddressCounters);
 
+/**
+ * This function sets the ICMPv6 socket on the infrastructure network interface.
+ *
+ * @param[in] aFd  The file descriptor of the ICMPv6 socket on the infrastructure network interface.
+ *
+ */
+void otSysInfraNetifSetIcmp6Socket(int aFd);
```

Comment:
How do we plan to support dynamic switch of AIL?

I suppose we need to provide a similar otSys* API for ot-daemon to init/deinit/enable/disable the Border Router components at runtime according to the AIL updates. If this is the case, we better design the otSys* able to support both cases. For example:

```
// If aIcmp6Socket is -1, a default one will be created in this method.
otSysReinitBorderRouterOnInfraIf(const char *aInfraIfName, int aIcmp6Socket);
```

Reply:
My current thought is to support switching AIL via OT core API instead `otSys*` API. Currently there're not many `otSys*` APIs changing the state of OT. When AIL switches, we may call 
`otBorderRoutingInit(instance, netifIndex, true)`.

It's not fully designed yet though.

Reply:
Actually I just found there may already be some support in switching the AIL: https://github.com/openthread/openthread/pull/9305.